lib/ui/window/pointer_data.h

```diff
@@ -9,6 +9,29 @@
 
 namespace flutter {
 
+// Must match the button constants in events.dart.
+namespace PointerButton {
+const int64_t kPrimary = 1 << 0;
+const int64_t kSecondary = 1 << 1;
```

Comment:
Are the raw values in event.dart actually used? I wouldn't expose them unless there's a need; it seemed like they were more for defining the device-specific constants.

Having them here encourages confusing usage (e.g., your use of `PointerButton::kPrimary` rather than `PointerButton::kMousePrimary` in this PR even though it's a mouse), and especially so given that they don't line up with the stylus names.

Reply:
Yes. In fact, the general series of buttons are used more often, because most widgets are defined for cross-device behaviors, and by grouping the same "cross-device behavior" into the same value, more often than not widgets do not need to check the device kind. For example, primary mouse button, stylus contact, and touch contact are supposed to behave the same for most cases, so they share the same value `0x01`, while secondary mouse button and primary stylus button share the same value `0x02`. In this way, a button click will respond to `kPrimaryButton`, instead of `kPrimaryMouseButton` or `kStylusContact`, while a context menu will respond to `kSecondaryButton`.

Reply:
I see, makes sense.

I still think we shouldn't expose them in the C++ code absent a reason though; we're creating device-specific data structs, rather than consuming data in a device-agnostic way. For creating them, using the device-specific values makes much more sense to me.

Reply:
I agree. However I still need a value when synthesizing the primary button. Currently I'm using `kPrimaryMouseButton` but it's a little awkward. I wonder if you think it's acceptable. https://github.com/flutter/engine/pull/8873/files#diff-3784eecbfe20523d62f0a4c19cd363ebR730

Reply:
Why is that awkward? The device type is hard-coded to mouse, so the primary mouse button seems like the right constant to me.